src/odemis/gui/cont/acquisition/fastem_acq.py

```diff
@@ -1342,14 +1437,32 @@ def on_run(self, evt):
             calib_1_done, calib_2_done, _ = self.get_calibration_status(focussed_view)
             if self._calib_1.IsChecked():
                 calib_names.append(CALIBRATION_1)
+                if not self._calib_1_vis_btn.GetValue():
+                    # show the calibration region again when the user starts the calibration
+                    calibration = current_sample.scintillators[scintillator_num].calibrations[CALIBRATION_1]
+                    calibration.shape.active.value = True
+                    calibration.shape.cnvs.add_world_overlay(calibration.shape)
+                    self._calib_1_vis_btn.SetValue(True)
             if self._calib_2.IsChecked() and (
                 CALIBRATION_1 in calib_names or calib_1_done
             ):
                 calib_names.append(CALIBRATION_2)
+                if not self._calib_2_vis_btn.GetValue():
+                    # show the calibration region again when the user starts the calibration
+                    calibration = current_sample.scintillators[scintillator_num].calibrations[CALIBRATION_2]
+                    calibration.shape.active.value = True
+                    calibration.shape.cnvs.add_world_overlay(calibration.shape)
+                    self._calib_2_vis_btn.SetValue(True)
             if self._calib_3.IsChecked() and (
                 CALIBRATION_2 in calib_names or calib_2_done
             ):
                 calib_names.append(CALIBRATION_3)
+                if not self._calib_3_vis_btn.GetValue():
+                    # show the calibration region again when the user starts the calibration
+                    calibration = current_sample.scintillators[scintillator_num].calibrations[CALIBRATION_3]
+                    calibration.shape.active.value = True
+                    calibration.shape.cnvs.add_world_overlay(calibration.shape)
```

Comment:
If it is possible to make the visible button as an attribute of the calibration, then this function can be generalized by making a new function. The new function takes calibration as input and set the attributes based on the visibility.

Reply:
The calibration is an instance of the `FastEMCalibration` class, which is part of the acquisition code, so it doesn't make sense to have the button as an attribute there. I will create a method on the current class, so I don't have to repeat the same code 3 times.